PHPCI/Command/DaemoniseCommand.php

```diff
@@ -65,46 +40,30 @@ protected function configure()
     */
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        $cmd = "echo %s > '%s/daemon/daemon.pid'";
-        $command = sprintf($cmd, getmypid(), PHPCI_DIR);
-        exec($command);
+        $this->setupLogging($input, $output);
 
-        $this->output = $output;
-        $this->run   = true;
-        $this->sleep = 0;
-        $runner      = new RunCommand($this->logger);
-        $runner->setMaxBuilds(1);
-        $runner->setDaemon(true);
+        file_put_contents(PHPCI_DIR."/daemon/daemon.pid", getmypid());
 
-        $emptyInput = new ArgvInput(array());
+        $maxSleep = (int)$input->getOption('interval');
 
-        while ($this->run) {
+        $run   = true;
+        $sleep = 0;
 
-            $buildCount = 0;
+        while ($run) {
+            $build = $this->findNextBuild();
 
-            try {
-                $buildCount = $runner->run($emptyInput, $output);
-            } catch (\Exception $e) {
-                $output->writeln('<error>Exception: ' . $e->getMessage() . '</error>');
-                $output->writeln('<error>Line: ' . $e->getLine() . ' - File: ' . $e->getFile() . '</error>');
+            if ($build) {
+                $this->runBuild($build);
+                $sleep = 0;
+                echo 'build';
```

Comment:
Just a quick one - Anywhere that you're currently using `echo`, can you please switch it over to using `$output->writeln(...)`